README.md

```diff
@@ -1 +1,409 @@
+        INCLUDE_ARG_DOC_PROCESSOR, // The @arg and @includeArg processor
+        COMMENT_DOC_PROCESSOR, // The @comment processor
+        SAMPLE_DOC_PROCESSOR, // The @sample and @sampleNoComments processor
+        "com.example.plugin.ExampleDocProcessor", // A custom processor, see below
+    )
+
+    // Optional dependencies for this task. These dependencies can introduce custom processors.
+    dependencies {
+        plugin("com.example:plugin:SOME_VERSION")
+    }
+}
+
+// Modify all Jar tasks such that before running the Kotlin sources are set to 
+// the target of processKdocMain and they are returned back to normal afterwards.
+tasks.withType<Jar> {
+    dependsOn(processKdocMain)
+    outputs.upToDateWhen { false }
+
+    doFirst {
+        kotlin {
+            sourceSets {
+                main {
+                    kotlin.setSrcDirs(processKdocMain.targets)
+                }
+            }
+        }
+    }
+
+    doLast {
+        kotlin {
+            sourceSets {
+                main {
+                    kotlin.setSrcDirs(kotlinMainSources)
+                }
+            }
+        }
+    }
+}
+
+..
+
+// As a bonus, this will update dokka if you use that
+tasks.withType<org.jetbrains.dokka.gradle.AbstractDokkaLeafTask> {
+    dokkaSourceSets {
+        all {
+            for (root in processKdocMain.targets)
+                sourceRoot(root)
+        }
+    }
+}
+```
+
+## @include Processor
+
+Adds the @include modifier to KDocs to reuse written docs.
+Anything you can target with `[target]` with KDoc can be included in the current KDoc and will replace the `@include` line with the other content one to one.
+Visibility modifiers are ignored for now. Import statements are taken into account, however!
+JavaDoc is also supported. Add the `"java"` extension to `fileExtensions` in the plugin setup to use it.
+You can even cross include between Java and Kotlin but no conversion whatsoever will be done at the moment.
+
+Example in conjunction with the `@arg` / `@includeArg` processor:
+```kotlin
+package com.example.plugin
+
+/**
+ * Hello World!
+ * 
+ * This is a large example of how the plugin will work from {@includeArg source}
+ * 
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin]
+ * {@arg source Test1}
+ */
+private interface Test1
+
+/**
+ * Hello World 2!
+ * @include [Test1]
+ * {@arg source Test2}
+ */
+@AnnotationTest(a = 24)
+private interface Test2
+
+/** 
+ * Some extra text
+ * @include [Test2]
+ * {@arg source someFun} */
+fun someFun() {
+    println("Hello World!")
+}
+
+/** {@include [com.example.plugin.Test2]}{@arg source someMoreFun} */
+fun someMoreFun() {
+    println("Hello World!")
+}
+```
+turns into:
+```kotlin
+package com.example.plugin
+
+/**
+ * Hello World!
+ *
+ * This is a large example of how the plugin will work from Test1
+ *
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin]
+ *
```

Comment:
Nope, on purpose. `{@arg source Test1}` is an inline tag that disappears after processing. However, since it's inline only the exact `{@..}` tag is removed, keeping the newline in place.